File: SkillBridgeSmsService/src/main/java/com/skillBridge/sms_service/service/ApplicationService.java
package com.skillBridge.sms_service.service;

import java.util.List;

import com.skillBridge.sms_service.dtos.ApplicationRequest;
import com.skillBridge.sms_service.dtos.ApplicationResponse;
import com.skillBridge.sms_service.dtos.ApplicationStatusUpdateRequest;
import com.skillBridge.sms_service.dtos.ApplicationUpdateRequest;

public interface ApplicationService {

    // ================= APPLY TO PROJECT =================
    ApplicationResponse create(ApplicationRequest request, Long studentId);

    // ================= READ =================
    ApplicationResponse get(Long applicationId, Long studentId);
    List<ApplicationResponse> getAll(Long studentId);

    // ================= UPDATE =================
    ApplicationResponse updateMessage(
            Long applicationId,
            ApplicationUpdateRequest request,
            Long studentId
    );

    // Only project owner can update status
    ApplicationResponse updateStatus(
            Long applicationId,
            ApplicationStatusUpdateRequest request,
            Long studentId
    );

    // ================= DELETE =================
    void delete(Long applicationId, Long studentId);

 // Creator/Admin: view applications for a project
    List<ApplicationResponse> getApplicationsForProject(Long projectId, Long requesterId);
}

File: SkillBridgeFrontend/src/services/applicationService.js
import api from './api';

export const applicationService = {
    apply: async (data) => {
        const response = await api.post('/applications/create', data);
        return response.data;
    },

    getById: async (id) => {
        const response = await api.get(`/applications/${id}`);
        return response.data;
    },

    getMyApplications: async () => {
        const response = await api.get('/applications');
        return response.data;
    },

    getForProject: async (projectId) => {
        const response = await api.get(`/applications/project/${projectId}`);
        return response.data;
    },

    updateMessage: async (id, data) => {
        const response = await api.patch(`/applications/update/${id}`, data);
        return response.data;
    },

    updateStatus: async (id, status) => {
        const response = await api.patch(`/applications/update/${id}/status`, { status });
        return response.data;
    },

    delete: async (id) => {
        const response = await api.delete(`/applications/delete/${id}`);
        return response.data;
    },
};

export default applicationService;
